python/mage/tgn/definitions/time_encoding.py

```diff
@@ -0,0 +1,23 @@
+import numpy as np
+import torch.nn as nn
+import torch
+
+
+# time encoding by GAT
+class TimeEncoder(nn.Module):
+    def __init__(self, out_dimension):
+        super().__init__()
+
+        self.out_dimension = out_dimension
+        self.w = nn.Linear(1, out_dimension)
+
+        self.w.weight = nn.Parameter(
+            (torch.from_numpy(1 / 10 ** np.linspace(0, 9, out_dimension)))
```

Comment:
What kind of initialization is this?

Reply:
This initialization E.Rossi used in his Temporal graph networks, so I used the same. I think there could be different types of initialization here.